packages/components/src/internal/components/chart/ChartColorInputs.tsx

```diff
@@ -83,6 +83,38 @@ function shapeValueRenderer(option) {
     return <ShapeOptionRenderer isValueRenderer name={option.data.value} />;
 }
 
+interface LineTypeOptionRendererProps {
+    isValueRenderer: boolean;
+    name: string;
+}
+
+// export for jest testing
+export const LineTypeOptionRenderer: FC<LineTypeOptionRendererProps> = memo(({ name, isValueRenderer }) => {
+    const dashedPath = <path d="M 0 5 H 30" stroke="#000000" stroke-width="3" stroke-dasharray="12, 3" fill="none" />;
+    const dottedPath = <path d="M 0 5 H 30" stroke="#000000" stroke-width="3" stroke-dasharray="2, 2" fill="none" />;
+    const solidPath = <path d="M 0 5 H 30" stroke="#000000" stroke-width="3" fill="none" />;
+
+    const className = classNames('chart-builder-type-option', { 'chart-builder-type-option--value': isValueRenderer });
+    return (
+        <span className={className} data-series-linetype={name}>
+            <svg width="30" height="10">
+                {name === 'dashed' && dashedPath}
+                {name === 'dotted' && dottedPath}
+                {name === 'solid' && solidPath}
+            </svg>
+        </span>
+    );
+});
```

Comment:
We can get rid of multiple definitions of largely the same DOM elements, and we can get rid of the conditional rendering for them. The linter was also saying that `stroke-dasharray` should be `strokeDasharray` and `stroke-width` should be `strokeWidth`.

```suggestion
const LINE_TYPE_STROKES = { dashed: '2, 2', dotted: '12, 3', solid: undefined };

interface LineTypeOptionRendererProps {
    isValueRenderer: boolean;
    name: string;
}

// export for jest testing
export const LineTypeOptionRenderer: FC<LineTypeOptionRendererProps> = memo(({ name, isValueRenderer }) => {
    const stroke = LINE_TYPE_STROKES[name];
    const className = classNames('chart-builder-type-option', { 'chart-builder-type-option--value': isValueRenderer });
    return (
        <span className={className} data-series-linetype={name}>
            <svg height="10" width="30">
                <path d="M 0 5 H 30" fill="none" stroke="#000000" strokeDasharray={stroke} strokeWidth="3" />
            </svg>
        </span>
    );
});
```

Reply:
done